### PR TITLE
Add ssl support

### PR DESCRIPTION
**Pull request by genotrance**

### CODE REVIEW

File: src/irc.nim [hunk 1]
```diff
@@ -333,6 +347,14 @@ proc newIrc*(address: string, port: Port = 6667.Port,
   result.userList = initTable[string, UserList]()
   result.eventsQueue = initDeque[IrcEvent]()
 
+  when defined(ssl):
+    if ssl:
+      try:
+        sslContext.wrapSocket(result.sock)
+      except:
+        result.sock.close()
+        raise getCurrentException()
```
Comment (Araq) [inline src/irc.nim]: ``raise`` without anything does re-raise the current exception and should be preferred.

File: src/irc.nim [hunk 2]
```diff
@@ -564,6 +586,8 @@ proc newAsyncIrc*(address: string, port: Port = 6667.Port,
               realname = "NimBot", serverPass = "",
               joinChans: seq[string] = @[],
               msgLimit: bool = true,
+              ssl: bool = false,
```
Comment (dom96) [inline src/irc.nim]: Call this ``useSsl``

File: src/irc.nim [hunk 3]
```diff
@@ -564,6 +586,8 @@ proc newAsyncIrc*(address: string, port: Port = 6667.Port,
...
+              sslContext = getDefaultSSL(),
               callback: proc (irc: AsyncIrc, ev: IrcEvent): Future[void]
```
Comment (dom96) [inline src/irc.nim:591]: Huh, I'm actually surprised that this is allowed (non-default argument after default arguments).

File: src/irc.nim [hunk 4]
```diff
@@ -590,6 +614,14 @@ proc newAsyncIrc*(address: string, port: Port = 6667.Port,
   result.status = SockIdle
   result.userList = initTable[string, UserList]()
 
+  when defined(ssl):
+    if ssl:
+      try:
+        sslContext.wrapSocket(result.sock)
+      except:
+        result.sock.close()
```
Comment (dom96) [inline src/irc.nim:622]: I think ``defer`` would work better here, can we use it?

**Review by dom96 — APPROVED** [review]
Couple of small things, but mostly looks good to merge.

File: src/irc.nim (hunk 4 above)
Comment (genotrance) [reply src/irc.nim:622]: I got this from httpclient. This is the constructor, why would we defer socket close here?

Comment (dom96) [reply src/irc.nim:622]: oh sorry, I'm being stupid. Never mind :)

**Pull request merged**